examples/android/CHIPTool/app/src/main/java/com/google/chip/chiptool/provisioning/DeviceProvisioningFragment.kt

```diff
@@ -87,8 +86,40 @@ class DeviceProvisioningFragment : Fragment() {
     val port = 5540
     val id = DeviceIdUtil.getNextAvailableId(requireContext())
     val deviceController = ChipClient.getDeviceController(requireContext())
+
     DeviceIdUtil.setNextAvailableId(requireContext(), id + 1)
     deviceController.setCompletionListener(ConnectionCallback())
+
+    deviceController.setDeviceAttestationDelegate(DEVICE_ATTESTATION_FAILED_TIMEOUT,
```

Comment:
This can create a memory leak (as the lambda holds a reference to the Fragment and Activity). Consider holding a reference to the `ChipController` and assigning in `onCreate()`, and then in `onDestroy()` set the attestation delegate back to null.

```
private lateinit var deviceController: ChipDeviceController

override fun onCreate(...) {
  super.onCreate(...)
  deviceController = ChipClient.getDeviceController(requireContext())
}

override fun onDestroy() {
  super.onDestroy()
  deviceController.close()
  deviceController.setDeviceAttestationDelegate(null)
}
```